forte/orbital-helpers/orbital_embedding.cc

```diff
@@ -834,37 +887,84 @@ std::shared_ptr<MOSpaceInfo> make_embedding(psi::SharedWavefunction ref_wfn,
     // Write a new MOSpaceInfo:
     std::map<std::string, std::vector<size_t>> mo_space_map;
 
-    // Frozen docc space
-    size_t freeze_o =
-        static_cast<size_t>(num_Fo + num_Bo + adj_sys_docc); // Add the additional frozen core to Bo
-    mo_space_map["FROZEN_DOCC"] = {freeze_o};
+    // Write orbital information into new mo_space_info based on different embedding methods:
 
-    // Restricted docc space
-    size_t ro = static_cast<size_t>(num_Ao - adj_sys_docc);
-    if (options->get_str("EMBEDDING_REFERENCE") == "HF") {
-        ro -= diff;
-    }
-    mo_space_map["RESTRICTED_DOCC"] = {ro};
+    // ASET(mf):
+    if (options->get_str("EMBEDDING_TYPE") == "ASET_MF") {
+        // Normal partition for Frozen-Core embedding
+        size_t freeze_o = static_cast<size_t>(num_Fo + num_Bo +
+                                              adj_sys_docc); // Add the additional frozen core to Bo
+        mo_space_map["FROZEN_DOCC"] = {freeze_o};
 
-    // Active space
-    size_t a = static_cast<size_t>(actv_a[0]);
-    if (options->get_str("EMBEDDING_REFERENCE") == "HF") {
-        a += diff;
-        a += diff2;
+        size_t ro = static_cast<size_t>(num_Ao - adj_sys_docc);
+        if (options->get_str("EMBEDDING_REFERENCE") == "HF") {
+            ro -= diff;
+        }
+        mo_space_map["RESTRICTED_DOCC"] = {ro};
+
+        size_t a = static_cast<size_t>(actv_a[0]);
+        if (options->get_str("EMBEDDING_REFERENCE") == "HF") {
+            a += diff;
+            a += diff2;
+        }
+        mo_space_map["ACTIVE"] = {a};
+
+        size_t rv = static_cast<size_t>(num_Av - adj_sys_uocc);
+        if (options->get_str("EMBEDDING_REFERENCE") == "HF") {
+            rv -= diff2;
+        }
+        mo_space_map["RESTRICTED_UOCC"] = {rv};
+
+        size_t freeze_v = static_cast<size_t>(
+            num_Fv + num_Bv + adj_sys_uocc); // Add the additional frozen virtual to Bv
+        mo_space_map["FROZEN_UOCC"] = {freeze_v};
     }
-    mo_space_map["ACTIVE"] = {a};
 
-    // Restricted uocc space
-    size_t rv = static_cast<size_t>(num_Av - adj_sys_uocc);
-    if (options->get_str("EMBEDDING_REFERENCE") == "HF") {
-        rv -= diff2;
+    // ASET(2):
+    if (options->get_str("EMBEDDING_TYPE") == "ASET2") {
+        outfile->Printf(
+            "\n\n  --------------- Generating Total (A + B) SpaceInfo --------------- ");
+        // This will make A->active, B->restricted, for multilayer embedding/downfolding tests
+        size_t freeze_o = static_cast<size_t>(num_Fo + adj_sys_docc);

```

Comment:
I don't think `static_cast` is needed as long as you specify the type explicitly for `freeze_o`.

Reply:
[Preferred C++ style](https://herbsutter.com/2013/06/13/gotw-93-solution-auto-variables-part-2/) would be `auto freeze_o = static_cast<size_t>(num_Fo + adj_sys_docc);` to explicitly signal that you are requesting a type conversion.

Reply:
This is fine with me. I'm just too lazy to follow the suggested style.